src/adapters/node.ts

```diff
@@ -68,7 +72,13 @@ class NodeServer implements Server {
       ...this.options.node,
     };
 
-    const server = NodeHttp.createServer(this.serveOptions, handler);
+    // Create HTTPS server if HTTPS options are provided, otherwise create HTTP server
+    const server = this.isHttps
+      ? NodeHttps.createServer(
+          { ...this.options.https, ...this.serveOptions },
```

Comment:
We can refactor this under `serveOptions`

Reply:
fixed